src/nncf/quantization/algorithms/weight_compression/algorithm.py

```diff
@@ -744,106 +791,111 @@ def apply(
-                    # Convolution ops are always kept in backup mode.
-                    # Embedding layers are quantized to 4-bits only if all_layers=True.
-                    # MatMul ops can't have multiple reduction axes.
-                    nncf_logger.warning(
-                        f"Weight compression expects a single reduction axis, but {len(reduction_axes)} given. "
-                        f"Weight shape: {weight_shape}, reduction axes: {reduction_axes}, "
-                        f"node name: {node.node_name}. The node will be in {self._backup_mode} mode."
-                    )
 
-                if self._backup_mode == BackupMode.NONE:
+                if is_target_node and self.is_weight_compression_supported(weight_dtype, self._mode):
+                    if (
+                        self._group_size != -1
+                        and self._all_layers
+                        and node.metatype in self._backend_entity.embedding_metatypes
+                        and isinstance(reduction_axes, tuple)
+                        and len(reduction_axes) != 1
+                    ):
+                        # NNCF supports multiple reduction axes only for ops with group_size != -1.
+                        # Convolution ops are always kept in backup mode.
+                        # Embedding layers are quantized to 4-bits only if all_layers=True.
+                        # MatMul ops can't have multiple reduction axes.
+                        nncf_logger.warning(
+                            f"Weight compression expects a single reduction axis, but {len(reduction_axes)} given. "
+                            f"Weight shape: {weight_shape}, reduction axes: {reduction_axes}, "
+                            f"node name: {node.node_name}. The node will be in {self._backup_mode} mode."
+                        )
+
                     wc_config = None
+                    if self._backup_mode != BackupMode.NONE:
+                        mode = (
+                            CompressWeightsMode.INT8_ASYM
+                            if self._backup_mode == BackupMode.INT8_ASYM
+                            else CompressWeightsMode.INT8_SYM
+                        )
+                        if self.is_weight_compression_supported(weight_dtype, mode):
+                            wc_config = WeightCompressionConfig(mode=mode)
+                    weight_params = WeightCompressionParameters(
+                        weight_name, node, weight_port_id, weight_dtype, weight_shape, reduction_axes, wc_config
+                    )
+                    all_weight_params.append(weight_params)
+                    weight_names.add(weight_name)
                 else:
-                    mode = (
-                        CompressWeightsMode.INT8_ASYM
-                        if self._backup_mode == BackupMode.INT8_ASYM
-                        else CompressWeightsMode.INT8_SYM
+                    is_last_layer_skipped = is_last_layer
```

Comment:
I have difficulties matching this logic with the previous one. Where does this come from / what's the logic behind it?